yapapi/rest/market.py

```diff
@@ -56,6 +57,15 @@ def __init__(self, *, _ref: models.Agreement):
         self.raw_details = _ref
 
 
+class AgreementState(statemachine.StateMachine):
```

Comment:
yes, but then, you need to access the AgreementPool -> and this enables this check independently from the AgreementPool

Reply:
OK, but this doesn't answer the question about the **exact** purpose.
If all we want to have is `terminated`, please add a `terminated` property and set in in a proper place.

This sort of a class is an additional complexity that makes code much harder to understand.
E.g. how much time would it take to answer question "why does the AgreementsPool - the class that manages agreements - doesn't care about AgreementState"?

IOW, this is a "pretty" solution that just simply makes `yapapi` worse.

Reply:
well, let me recall a conversation that I had with you on another pull request some time ago... 

I added some method or a property that served a single, well defined purpose

You told me this was bad design because it was incidental that we needed this particular method now and our solution should be more futureproof and should solve all similar needs not this particular that I had in that very moment... 

So here it is - _now_, yes, I need to check `terminated` particularly... but I'd like to have a solution that allows me to check the other possible states as well without adding arbitrary flags each time we suddently find something useful...

Reply:
I remember that conversation.

1. In **this** case we're talking about a class that is not a part of a public interface (at least, it's not in the API Reference). Then, it was about `Cluster` - the most important new part of the interface.
2. In **this** case I'm against increasing the complexity of `yapapi`. This is exactly the same thing I wanted then (at a cost of making examples a little less readable).
3. If one day we'll decide we need more states then "terminated?", it will be trivial to move statemachine. But until then, this additional class is just additional weight.
4. Because once we didn't have this class and still needed such logic, the code is harder to understand, since we have the same logic implemented in different ways. E.g. `AgreementsPool` doesn't care about `agreement.confirmed`, just uses the return value of `agreement.confirm()`. Not pretty.

In general, from my POV `yapapi` suffers from unused things that are supposed to be "pretty" or "general", but they are never used and just make it harder to understand and maintain.

And as an additional point: if you

Reply:
> One more thing to think about: are you sure `yapapi` should track agreement state? I think that `yagna` should be responsible for this. What if the other side terminates the agreement?

on one hand yes, and on another, your changes did verify the status in yagna but rather relied on an internal state of the agreements pool ...

Reply:
ok